wwwbase/js/scramble.js

```diff
@@ -0,0 +1,280 @@
+$(document).ready(function() {
+  callAjax(); // init AJAX
+  selectDifficulty();
+  inputListen();
+  checkWord();
+  testing();
+});
+
+function callAjax()
+{
+  result;
+   $.ajax({
+      type: "POST",
+      url: wwwRoot + "ajax/scramble.php",
+    })
+    .done(function(response) {
+      result = $.parseJSON(response);
+    })
+    .fail(function() {
+      console.log("Nu merge");
+    });
+  console.log(result);
+  return result;
+}
+
+var score = 0;
+var cnt = 0;
+var searchWord;
+var lettersPressed = new String(); // in acest array se retin literele tastate
+var totalWords = new Array(); // the possible words that can be made from the randomWord.
+function selectDifficulty() {
+  $(".difficultyButton").on("click", function() {
+    // this e pentru a prelua valoarea butonului tocmai apasat, si nu a unuia oarecare
+    var difficulty = $(this).attr("value");
+      
+      var word = callAjax();
+      if(result.everyWord.length < 4) {
+      word = callAjax();
+      }
+      cnt = 0;
+      lettersPressed = [];
+      totalWords = word.everyWord;
+      $("#result").html(word.randomWord);
+      $("#maxWords").html(word.everyWord.length);
+      drawLetters(word.randomWord);
+      startTimer(difficulty);
+      console.log(totalWords);
+      console.log(word.randomWord);
+  });
+}
+
+hide = 0;
+function testing() {
+$(".wordBtn").on("click", function() {
+  console.log("pushed");
+  	if ( !hide ) {
+      $(".wordArea").html(totalWords + " ");
+      hide = 1;
+	} else {
+      $(".wordArea").html("");
+      hide = 0;
+	}
+  });
+}
+
+function inputListen() {
+  layers = $("canvas").getLayers();
+
+  $(document).keyup(function(letter) {
+    //var searchWord = $(this).val();
+    var key;
+    key = letter.keyCode;
+    var keyString;
+    keyString = String.fromCharCode(key);
+
+    for(var i = 0; i< layers.length; i += 2) {
+      if(layers[i].data.letter == "\u00c2" || layers[i].data.letter == "\u0102") {
+        layers[i].data.letter = "A";
+      }
+      if(layers[i].data.letter == "\u00ee" || layers[i].data.letter == "\u00ce") {
+        layers[i].data.letter = "I";
+      }
+      if(layers[i].data.letter == "\u0219" || layers[i].data.letter == "\u0218") {
+        layers[i].data.letter = "S";
+      }
+      if(layers[i].data.letter == "\u021b" || layers[i].data.letter == "\u021a") {
+        layers[i].data.letter = "T";
+      }
+    }
```

Comment:
Asta nu e fantastic. Diacriticele sunt importante și sunt destui oameni care le folosesc. De asemenea, am dat peste un bug (sau cel puțin, un comportament neintuitiv) unde tasta „Ă” mișca două litere simultan, „A” și „Ă”.

Nu mai știu cum am rezolvat problema, dar vezi jocul _Spânzurătoarea_ (`wwwbase/js/hangman.js`), care poate fi jucat și cu diacritice.